### PR TITLE
[PIR] Fix some pir interpreter bug and refine some code

### PR DESCRIPTION
**Pull request by zhangbo9674**
### PR types

Bug fixes
### PR changes

Others
### Description

Move the PIR control flow related instructions to the control_flow dir;
Fix some pir interpreter bug and refine some code
Pcard-67164

### CODE REVIEW

File: paddle/fluid/framework/new_executor/pir_interpreter.cc [hunk 1]
```diff
@@ -48,16 +48,16 @@
...
 #include "paddle/fluid/framework/new_executor/instruction/builtin_combine_instruction.h"
-#include "paddle/fluid/framework/new_executor/instruction/has_elements_instruction.h"
-#include "paddle/fluid/framework/new_executor/instruction/if_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/assert_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/has_elements_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/if_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/select_input_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/tuple_pop_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/tuple_push_instruction.h"
+#include "paddle/fluid/framework/new_executor/instruction/control_flow/while_instruction.h"
```
Comment (Aurelius84) [inline paddle/fluid/framework/new_executor/pir_interpreter.cc:58]: 后续我们是否可以在`instruction/control_flow/` 加个`all.h` 头文件，这样后续就可直接用这个了？

**Review by Aurelius84 — APPROVED** [review]

File: paddle/fluid/framework/new_executor/pir_interpreter.cc (hunk 1 above)
Comment (zhangbo9674) [reply paddle/fluid/framework/new_executor/pir_interpreter.cc:58]: 好主意，单独提 PR 完善一下

File: paddle/fluid/ir_adaptor/translator/program_translator.cc [hunk 2]
```diff
@@ -419,6 +421,34 @@ inline pir::Operation* InsertSetParamaterOp(pir::IrContext* ctx,
...
 }
 
+void ProgramTranslator::InsertDataOpForSingleBlock(const BlockDesc& block) {
+  std::unordered_set<std::string> all_var_names;
+  for (auto& var : block.AllVars()) {
+    all_var_names.insert(var->Name());
+  }
+
+  std::unordered_set<std::string> inner_outputs;
+  for (auto op_desc : block.AllOps()) {
```
Comment (chen2016013) [inline paddle/fluid/ir_adaptor/translator/program_translator.cc:431]: `auto& op_desc`

**Review by chen2016013 — APPROVED** [review]
LGTM

**Review by kangguangli — APPROVED** [review]

**Review by From00 — APPROVED** [review]

**Pull request merged**